pkg/kubelet/prober/results/results_manager.go

```diff
@@ -40,14 +40,17 @@ type Manager interface {
 }
 
 // Result is the type for probe results.
-type Result bool
+type Result int
 
 const (
-	// Success is encoded as "true" (type Result)
-	Success Result = true
+	// Success is encoded as 0 (type Result)
+	Success Result = 0
 
-	// Failure is encoded as "false" (type Result)
-	Failure Result = false
+	// Failure is encoded as 1 (type Result)
+	Failure Result = 1
+
+	// Unknown is encoded as -1 (type Result)
+	Unknown Result = -1
 )
 
 func (r Result) String() string {
```

Comment:
Should we add the new `Result` type here (and in `ToPrometheusType`) as well? I think the current behavior is ok, reporting -1/"UNKNOWN", but it may be nice to be explicit. Ill defer that to the approvers :smile: